File: user/m_files/atan_full.m
function theta = atan_full( opp, adj )
% atan_full -- arctangent(opp/adj) that tests inputs & adjusts for quadrant to return answer in [-180 +180]
% Mark J. Duvall ~ mjduvall@hawaii.edu ~ October 2015 ~ %

% opp = "opposite" (perpendicular); adj = "adjacent" (parallel)

if adj >= 0
  theta = atan( opp / adj );
else
  if opp >= 0
    theta = atan( opp / adj ) + pi;
  else
    theta = atan( opp / adj ) - pi;
  end
end

end %function

File: user/root_macros/aTanFull.cxx
// aTanFull.cxx -- arctangent(opp/adj) that tests inputs & adjusts for quadrant to return answer in [-180 +180]
// Mark J. Duvall ~ mjduvall@hawaii.edu ~ 3/2020 ~ //
// opp = "opposite" (perpendicular); adj = "adjacent" (parallel)

double aTanFull( double opp, double adj ) {

Double_t phi;

if ( adj >= 0 ) {
  phi = TMath::ATan( opp / adj );
} else {
  if ( opp >= 0 ) {
    phi = TMath::ATan( opp / adj ) + TMath::Pi();
  } else {
    phi = TMath::ATan( opp / adj ) - TMath::Pi();
  }
}

// all pau!
return phi;
}
